solutions/src/test/java/ru/sberbank/school/task06/CollectionUtilsTest.java

```diff
@@ -0,0 +1,214 @@
+        List<Plant> plants = new ArrayList<>();
+        plants.add(new Tree());
+        plants.add(new Maple());
+        plants.add(new Oak());
+
+        List<Tree> contains = new ArrayList<>();
+        Tree tree1 = (Tree) plants.get(0);
+        Maple tree2 = (Maple) plants.get(1);
+        add(contains, tree1);
+        add(contains, tree2);
+        add(contains, new Oak());
+
+        boolean containsAll = containsAll(plants, contains);
+        assertFalse(containsAll);
+    }
+
+    @Test
+    public void containsAnyTestTrue() {
+        List<Plant> plants = new ArrayList<>();
+        plants.add(new Tree());
+        plants.add(new Maple());
+        plants.add(new Oak());
+
+        List<Tree> contains = new ArrayList<>();
+        Tree tree1 = (Tree) plants.get(0);
+        add(contains, tree1);
+        add(contains, new Maple());
+        add(contains, new Oak());
+
+        boolean containsAny = containsAny(plants, contains);
+        assertTrue(containsAny);
+    }
+
+    @Test
+    public void containsAnyTestFalse() {
+        List<Plant> plants = new ArrayList<>();
+        plants.add(new Tree());
+        plants.add(new Maple());
+        plants.add(new Oak());
+
+        List<Tree> contains = new ArrayList<>();
+        add(contains, new Maple());
+        add(contains, new Oak());
+
+        boolean containsAny = containsAny(plants, contains);
+        assertFalse(containsAny);
+    }
+
+    @Test
+    public void rangeTestComparable() {
+        List<Integer> target = Arrays.asList(3, 4, 5, 6);
+        List<Integer> list = range(Arrays.asList(8, 1, 3, 5, 6, 4), 3, 6);
+        assertThat(list, equalTo(target));
+    }
+
+    @Test
+    public void rangeTestComparator() {
+        Comparator<Plant> comparator = new Comparator<Plant>() {
+            @Override
+            public int compare(Plant o1, Plant o2) {
+                String info1 = o1.getInfo();
+                String info2 = o2.getInfo();
+                if (info1.compareTo(info2) > 0) return 1;
+                if (info1.compareTo(info2) < 0) return -1;
+                return 0;
+            }
+        };
+
+        List<Plant> plants = new ArrayList<>();
+        plants.add(new Oak());//0
+        plants.add(new Tree());//1
+        plants.add(new Oak());//2
+        plants.add(new Maple());//3
+        plants.add(new Plant());//4
+        plants.add(new Oak());//5
+
+        List<Plant> target = new ArrayList<>();
+        add(target, plants.get(4));
+        add(target, plants.get(1));
+
+        List<Plant> result = range(plants, new Plant(), new Tree(), comparator);
+        System.out.println(result.toString());
+        assertThat(result, equalTo(target));
+    }
+
+    @Getter
+    class Plant {
+        String info = "1 level";
+    }
+    @Getter
+    class Tree extends Plant {
+        String info = "2 level";
+    }
+    @Getter
+    class Oak extends Tree {
+        String info = "3 level";
+    }
+    @Getter
```

Comment:
&#39;CLASS_DEF&#39; should be separated from previous statement.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a79175014>*